File: Backend-RS/models/intereses.py
from sqlmodel import Field, SQLModel, Relationship
from typing import List
from typing import Optional

from config.database import get_session
class InteresBase(SQLModel):
    Nombre: str = Field(index=True)  # si quieres hacerlo único: Field(index=True, unique=True)

class Intereses(InteresBase, table=True):
    IdInteres: Optional[int] = Field(default=None, primary_key=True)
    perfiles_rel: List["PerfilInteres"] = Relationship(back_populates="interes")

class InteresCreate(InteresBase):
    pass

class InteresPublic(InteresBase):
    IdInteres: int

class InteresUpdate(InteresBase):
    pass

InteresPublic.model_rebuild()

File: frontendrs/lib/data/models/intereses.dart
class Interes {
  final int idInteres;
  final String nombre;

  const Interes({required this.idInteres, required this.nombre});

  factory Interes.fromJson(Map<String, dynamic> json) {
    return Interes(
      idInteres: json['IdInteres'] ?? 0,
      nombre: json['Nombre'] ?? "",
    );
  }

  Map<String, dynamic> toJson() => {'IdInteres': idInteres, 'Nombre': nombre};
}

class InteresCreate {
  final String nombre;

  const InteresCreate({required this.nombre});

  Map<String, dynamic> toJson() => {'Nombre': nombre};
}

class InteresUpdate {
  final String nombre;

  const InteresUpdate({required this.nombre});

  Map<String, dynamic> toJson() => {'Nombre': nombre};
}
